docs/examples/plot_interval_transposition_error.py

```diff
@@ -0,0 +1,157 @@
+# the corresponding POA irradiance.  At such a short timescale, the
+# difference between instantaneous and interval-averaged irradiance
+# is negligible.
+
+# baseline: all calculations done at 1-second scale
+location = pvlib.location.Location(40, -80, tz='Etc/GMT+5')
+times = pd.date_range('2019-06-01 05:00', '2019-06-01 19:00',
+                      freq='1s', tz='Etc/GMT+5')
+solpos = location.get_solarposition(times)
+clearsky = location.get_clearsky(times, solar_position=solpos)
+poa_1s = transpose(clearsky, timeshift=0)  # no shift needed for 1s data
+
+# %%
+# Now, we will aggregate the 1-second values into interval averages.
+# To see how the averaging interval affects results, we'll loop over
+# a few common data intervals and accumulate the results.
+
+results = []
+
+for timescale_minutes in [1, 5, 10, 15, 30, 60]:
+
+    timescale_str = f'{timescale_minutes}min'
+    # get the "true" interval average of poa as the baseline for comparison
+    poa_avg = poa_1s.resample(timescale_str).mean()
+    # get interval averages of irradiance components to use for transposition
+    clearsky_avg = clearsky.resample(timescale_str).mean()
+
+    # low-res interval averages of 1-second data, with NO shift
+    poa_avg_noshift = transpose(clearsky_avg, timeshift=0)
+
+    # low-res interval averages of 1-second data, with half-interval shift
+    poa_avg_halfshift = transpose(clearsky_avg, timeshift=timescale_minutes/2)
+
+    df = pd.DataFrame({
+        'ground truth': poa_avg,
+        'modeled, no shift': poa_avg_noshift,
+        'modeled, half shift': poa_avg_halfshift,
+    })
+    # calculate error statistics and save for later
+    error = df.subtract(df['ground truth'], axis=0)
+    stats = error.abs().mean()
+    stats['timescale_minutes'] = timescale_minutes
+    results.append(stats)
+
+df_results = pd.DataFrame(results).set_index('timescale_minutes')
+print(df_results)
+
+# %%
+# The differences shown above are the absolute difference in :math:`W/m^2`.
+# In this example, using the timestamps unadjusted creates an error that
+# increases with increasing interval length, up to a ~40% :math:`W/m^2` error
+# at hourly resolution.  In contrast, incorporating a half-interval shift
+# so that solar position is calculated in the middle of the interval
+# instead of the edge reduces the error by one or two orders of magnitude:
+
+fig, ax = plt.subplots(figsize=(5, 3))
+df_results[['modeled, no shift', 'modeled, half shift']].plot.bar(rot=0, ax=ax)
+plt.ylabel('Mean Absolute Error [W/m$^2$]')
+plt.xlabel('Transposition Timescale [minutes]')
+fig.tight_layout()
+
+# %%
+# We can also plot the underlying time series results of the last
+# iteration (hourly in this case).  The modeled irradiance using
+# no shift is effectively time-lagged compared with ground truth.
+# In contrast, the half-shift model is nearly identical to the ground
+# truth irradiance.
+
+fig, ax = plt.subplots(figsize=(5, 3))
+ax = df.plot(ax=ax)
```

Comment:
I think it would be helpful if you added the `styles` kwarg to make the `modeled, half shift` dashed.